### PR TITLE
Use react native hcaptcha drawer

### PR DESCRIPTION
**Pull request by sddioulde**
### Description

Screen Shot 2021-11-19 at 2 55 38 PM

### Dragons

Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.

### How will this change be monitored?

For features that are critical or could fail silently please describe the monitoring/alerting being added.

### CODE REVIEW

File: src/containers/audio-rewards-page/store/mobileSagas.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { takeEvery, put } from 'redux-saga/effects'
+
+import { updateHCaptchaScore } from 'containers/audio-rewards-page/store/slice'
+import { MessageType } from 'services/native-mobile-interface/types'
+
+function* watchUpdateHCaptchaScore() {
+  yield takeEvery(MessageType.UPDATE_HCAPTCHA_SCORE, function* (action: {
```
Comment (raymondjacobson) [inline src/containers/audio-rewards-page/store/mobileSagas.ts:7]: we should be able to do this without explicitly passing the message using the common client stuff right? @sliptype

Comment (sliptype) [reply src/containers/audio-rewards-page/store/mobileSagas.ts:7]: No this is correct, the store only syncs in one direction. Passing data from the mobile client to the client still needs to be done with message passing. When everything is implemented in RN there will no longer be a need for message passing

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/sd-hcaptcha-drawer

File: src/common/store/pages/audio-rewards/slice.ts [hunk 2]
```diff
@@ -127,33 +129,8 @@ export const {
   resetClaimStatus,
   setHCaptchaStatus,
   resetHCaptchaStatus,
+  updateHCaptchaScore,
   setCognitoFlowStatus
 } = slice.actions
```
Comment (raymondjacobson) [inline src/common/store/pages/audio-rewards/slice.ts:135]: ty for moving

File: src/services/native-mobile-interface/types.ts [hunk 3]
```diff
@@ -122,7 +122,10 @@ export enum MessageType {
   // Theme
   THEME_CHANGE = 'theme-change',
 
-  SYNC_CLIENT_STORE = 'sync-client-store'
+  SYNC_CLIENT_STORE = 'sync-client-store',
+
+  // hCaptcha
+  UPDATE_HCAPTCHA_SCORE = 'UPDATE_HCAPTCHA_SCORE'
```
Comment (raymondjacobson) [inline src/services/native-mobile-interface/types.ts:128]: update-hcaptcha-score for consistency

File: src/containers/audio-rewards-page/store/mobileSagas.ts (hunk 1 above)
Comment (raymondjacobson) [reply src/containers/audio-rewards-page/store/mobileSagas.ts:7]: 👍  yes yes

**Review by raymondjacobson — APPROVED** [review]

File: src/containers/audio-rewards-page/components/ClaimRewardButton.tsx [hunk 4]
```diff
@@ -5,29 +5,29 @@ import { useDispatch, useSelector } from 'react-redux'
...
-} from '../store/slice'
 import {
   ChallengeRewardID,
   FailureReason,
   FlowErrorEvent,
   FlowSessionEvent,
   FlowUICloseEvent,
   FlowUIOpenEvent
-} from '../types'
+} from '../../../common/models/AudioRewards'
```
Comment (sliptype) [inline src/containers/audio-rewards-page/components/ClaimRewardButton.tsx:23]: Might be nicer to use absolute paths here
```suggestion
} from 'common/models/AudioRewards'
```

**Review by sliptype — APPROVED** [review]
Looks good!

**Pull request merged**